block/manager.go

```diff
@@ -449,13 +462,37 @@ func (m *Manager) getCommit(header types.Header) (*types.Commit, error) {
 	}, nil
 }
 
+func (m *Manager) IsProposer() (bool, error) {
+	// if proposer is not set, assume self proposer
+	if m.lastState.Validators.Proposer == nil {
```

Comment:
Consider moving mutex lock / unlock around read here.

Reply:
it is kind of same as the function itself has multiple reads to `m.lastState` one at the beginning and one at the end. will it make any difference?